File: VB/WinWebSolution.Module/Controllers/MasterDetailViewControllerBase.vb
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Editors

Namespace WinWebSolution.Module

    Public Class MasterDetailViewControllerBase
        Inherits ViewController

        Public Sub New()
            TargetViewType = ViewType.DetailView
            TargetViewNesting = Nesting.Root
        End Sub

        Protected Overrides Sub OnActivated()
            MyBase.OnActivated()
            For Each lpe As ListPropertyEditor In CType(View, DetailView).GetItems(Of ListPropertyEditor)()
                If lpe.Frame IsNot Nothing Then
                    CustomizeMasterDetailViewInfo(lpe.Frame)
                Else
                    AddHandler lpe.ControlCreated, New EventHandler(Of EventArgs)(AddressOf lpe_ControlCreated)
                End If
            Next
        End Sub

        Private Sub lpe_ControlCreated(ByVal sender As Object, ByVal e As EventArgs)
            CustomizeMasterDetailViewInfo(CType(sender, ListPropertyEditor).Frame)
        End Sub

        Private Sub CustomizeMasterDetailViewInfo(ByVal frame As Frame)
            For Each c As Controller In frame.Controllers
                If TypeOf c Is IMasterDetailViewInfo Then
                    CType(c, IMasterDetailViewInfo).AssignMasterDetailViewId(View.Id)
                End If
            Next
        End Sub
    End Class
End Namespace

File: CS/WinWebSolution.Module/Controllers/MasterDetailViewControllerBase.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Editors;

namespace WinWebSolution.Module {
    public class MasterDetailViewControllerBase : ViewController {
        public MasterDetailViewControllerBase() {
            TargetViewType = ViewType.DetailView;
            TargetViewNesting = Nesting.Root;
        }
        protected override void OnActivated() {
            base.OnActivated();
            foreach (ListPropertyEditor lpe in ((DetailView)View).GetItems<ListPropertyEditor>()) {
                if (lpe.Frame != null) {
                    CustomizeMasterDetailViewInfo(lpe.Frame);
                }
                else {
                    lpe.ControlCreated += new EventHandler<EventArgs>(lpe_ControlCreated);
                }
            }
        }
        void lpe_ControlCreated(object sender, EventArgs e) {
            CustomizeMasterDetailViewInfo(((ListPropertyEditor)sender).Frame);
        }
        private void CustomizeMasterDetailViewInfo(Frame frame) {
            foreach (Controller c in frame.Controllers) {
                if (c is IMasterDetailViewInfo) {
                    ((IMasterDetailViewInfo)c).AssignMasterDetailViewId(View.Id);
                }
            }
        }
    }
}